src/nav/nav.ts

```diff
@@ -0,0 +1,287 @@
+import {
+  AfterContentChecked,
+  AfterContentInit,
+  Attribute,
+  ChangeDetectorRef,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  QueryList,
+  SimpleChanges,
+  TemplateRef
+} from '@angular/core';
+import {isDefined} from '../util/util';
+import {NgbNavConfig} from './nav-config';
+
+let navCounter = 0;
+
+/**
+ * Context passed to the nav content template.
+ *
+ * See [this demo](#/components/nav/examples#keep-content) as the example.
+ */
+export interface NgbNavContentContext {
+  /**
+   * If `true`, current nav content is visible and active
+   */
+  $implicit: boolean;
+}
+
+
+/**
+ * This directive must be used to wrap content to be displayed in the nav.
+ */
+@Directive({selector: 'ng-template[ngbNavContent]'})
+export class NgbNavContent {
+  constructor(public templateRef: TemplateRef<any>) {}
+}
+
+
+/**
+ * The directive used to group nav link and related nav content. As well as set nav identifier and some options.
+ */
+@Directive({selector: '[ngbNavItem]', exportAs: 'ngbNavItem', host: {'class': 'nav-item'}})
```

Comment:
I think we should use `'[class.nav-item]': 'true'` for the class binding to avoid any collision and/or override